extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerMemoryParameters.java

```diff
@@ -282,13 +283,19 @@ private static long memoryPerBundle(
   )
   {
     final int bundleCount = numWorkersInJvm + numProcessingThreadsInJvm;
-    return (long) (maxMemoryInJvm * USABLE_MEMORY_FRACTION) / bundleCount;
+
+    // Need to subtract memoryForStatisticsTracking off the top, since this is reserved for statistics collection.
+    final long memoryForStatisticsTracking = (long) numWorkersInJvm * StageDefinition.PARTITION_STATS_MAX_BYTES;
+    final long memoryForBundles = (long) (maxMemoryInJvm * USABLE_MEMORY_FRACTION - memoryForStatisticsTracking);
```

Comment:
Should we add a > 0 check here? That way the user knows that he needs to give more memory to the task

Reply:
There's a reasonable error that occurs later on; see WorkerMemoryParametersTest.